File: tests/XMP/global-view/structure/F/shadow_reflect.f90
program shadow_reflect
  implicit none
  type b
     real(8):: a(10)
  end type b
!$xmp nodes p(*)
!$xmp template t(10)
!$xmp distribute t(block) onto p
!$xmp align b%a(i) with t(i)
!$xmp shadow b%a(1)
  type(b):: c
  integer :: i
  logical :: flag = .false.

!$xmp loop on t(i)
  do i=1, 10
     c%a(i) = i;
  end do
!$xmp reflect (c%a)
  
  if (xmp_node_num() == 1) then
     if (c%a(6) == 6) then
        flag = .true.
     end if
  else
     if (c%a(5) == 5) then
        flag = .true.
     end if
  end if

!$xmp reduction(.and.:flag)
  
  if (flag .eqv. .true.) then
    if (xmp_node_num() == 1) then
       write(*,*) "OK"
    end if
 else
    if (xmp_node_num() == 1) then
       write(*,*) "Error!"
    end if
    call exit(1)
 end if
 
end program shadow_reflect

File: tests/XMP/global-view/structure/C/shadow_reflect.c
#include <stdio.h>
#include <stdlib.h>
#include <xmp.h>
#define XMP_FALSE 0
#define XMP_TRUE  1
#pragma xmp nodes p[*]
#pragma xmp template t[10]
#pragma xmp distribute t[block] onto p
struct test{
  int a[10];
};
#pragma xmp align test.a[i] with t[i]
#pragma xmp shadow test.a[1]
struct test c;

int main(){
#pragma xmp loop on t[i]
  for(int i=0;i<10;i++)
    c.a[i] = i;

  int flag = XMP_TRUE;
#pragma xmp reflect (c.a)

  if(xmpc_node_num() == 0){
    if(c.a[5] != 5)
      flag = XMP_FALSE;
  }
  else{
    if(c.a[4] != 4)
      flag = XMP_FALSE;
  }

#pragma xmp reduction(*:flag)
  if(flag == XMP_TRUE){
    if(xmpc_node_num() == 0) printf("OK\n");
  }
  else{
    if(xmpc_node_num() == 0) printf("Error !\n");
    exit(1);
  }

  return 0;
}
